client/state/document-head/reducer.js

```diff
@@ -18,7 +18,7 @@ import {
 export function title( state = '', action ) {
 	switch ( action.type ) {
 		case DOCUMENT_HEAD_TITLE_SET:
-			return action.title;
+			return action.title || state;
```

Comment:
Hrm, I forgot about this part

> there are cases where we want to intentionally reset the title to `undefined` (#5897)

Anyway, we just can't dispatch `title` being set to `undefined`, or we'll get [that error](https://github.com/Automattic/wp-calypso/pull/7600#r76866662).